src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs

```diff
@@ -68,32 +79,40 @@ internal static partial class ConflictResolver
                 }
             }
 
-            return await ResolveConflictsInCurrentProcessAsync(
-                renameLocationSet, replacementText, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
+            var heavyweightLocations = await lightweightRenameLocations.ToHeavyweightAsync(cancellationToken).ConfigureAwait(false);
+            if (heavyweightLocations is null)
+                return new ConflictResolution(WorkspacesResources.Failed_to_resolve_rename_conflicts);
+
+            return await ResolveHeavyweightConflictsInCurrentProcessAsync(
+                heavyweightLocations, replacementText, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
```

Comment:
once we've marshalled over to OOP, we convert hte lightweight information back to hte heavyweight information needed to do the processing.  This should be cheap on the OOP side as the solutino/compilaitons are already synced and available there.